### DISCUSSION TITLE
GitHub Actions environment variable not set with input

### DISCUSSION THREAD

**Discussion by paramt** [community/community#25600] [Actions]
<p>In my <a href="https://github.com/paramt/github-actions-playground/blob/master/action.yml" rel="nofollow noopener">action.yml</a> I defined an input:</p>
<pre><code>name: 'test action'
author: Param Thakkar
description: 'test'

inputs:
  test_var:
    description: 'A test variable'
    required: true

runs:
  using: 'docker'
  image: 'Dockerfile'
</code></pre>
<p>And in my workflow I passed the test_var:</p>
<pre><code>name: CI

on: [push]

jobs:
  build:

    runs-on: ubuntu-latest

    steps:
      - name: Test the GH action
        uses: paramt/github-actions-playground@master
        with:
          test_var: "this is just a test"
</code></pre>
<p>So there should be an environment variable that’s created when the workflow runs, right? But when I run this short python script:</p>
<pre><code>import os

print(os.getenv('TEST_VAR'))
print("It works!")

exit(0)
</code></pre>
<p>It prints:</p>
<pre><code>None
It works!
</code></pre>
<p>Do I have to pass the env variable to my python script through the Dockerfile? Right now my Dockerfile looks like this: </p>
<pre><code>FROM python:latest

ADD entrypoint.py /entrypoint.py
ADD requirements.txt /requirements.txt

RUN echo "$TEST_VAR"

RUN pip install -r requirements.txt
RUN chmod +x entrypoint.py
ENTRYPOINT ["/entrypoint.py"]
</code></pre>
<p>Any help is appreciated. Here’s <a href="https://github.com/paramt/github-actions-playground" rel="nofollow noopener">the link to the repo</a></p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/paramt">@paramt</a>,</p>
<p>Thank you for being here and apologies for the delay in response, I’m wondering if you are still having this issue? I think it might be a syntax problem on the variable name.</p>

**Comment by kellervater** [comment]
<p>Yup, got the same problem in a different manifestation.</p>
<p>I followed the manual <a href="https://help.github.com/en/github/automating-your-workflow-with-github-actions/creating-a-docker-container-action" rel="nofollow noopener">https://help.github.com/en/github/automating-your-workflow-with-github-actions/creating-a-docker-container-action</a></p>
<p>but altered a few details to get it working for my purpose. <br>
My Use Case: Execute Ansible Playbooks to execute Deployments. The name of the Ansible Playbook is the input:</p>
<pre><code># action.yml
name: 'Ansible Playbook'
description: 'Executes an Ansible Playbook as Github Action'
inputs:
  playbookName:
    description: 'Name of the playbook in your workspace.'
    required: true
    default: 'Playbook-Name-Not-Set'
runs:
  using: 'docker'
  image: 'Dockerfile'
  args:
    - ${{ inputs.playbookName }}
</code></pre>
<p>Here’s the <code>entrypoint.sh</code></p>
<pre><code>#!/bin/sh

ansible-playbook "$INPUT_PLAYBOOKNAME"
</code></pre>
<p>Here’s the example workflow to test the whole thing:</p>
<pre><code>on: [push]

jobs:
  ansible_playbook_job:
    runs-on: ubuntu-latest
    name: A job to execute a sample playbook
    steps:
      - uses: actions/checkout@v1
      - uses: ./
      - name: Execute Playbook Action Step
        id: Execution
        uses: myorga/ansible-playbook-docker-action@master
        with:
          playbookName: "simple-playbook.yml"
</code></pre>
<p>Here’s my output:</p>
<pre><code>ERROR! the playbook: Playbook-Name-Not-Set could not be found
</code></pre>
<p>You can see it’s using the default value defined in the actions metadata.</p>
<p>Even if i try to use the variable as <code>$1</code> I get the same result.</p>
<p>And here’s the thing: If i don’t default the value it doesn’t fail. There’s just an empty output like this:</p>
<pre><code>ERROR! the playbook: could not be found
</code></pre>
<p>Maybe there’s something wrong in my syntax. <br>
Any suggestions?</p>

**Comment by kellervater** [comment]
<p>Ok, Solved it myself. It was a Problem with cribbing the whole manual and adapting it wrongly.<br>
<img src="https://user-images.githubusercontent.com/33324812/181091961-8c7cd65f-2bac-4485-8581-1ca1085fa51f.png" alt="wrong_action.PNG" data-base62-sha1="7W8DQjrP3n8udDffowfm4VHvK4h" width="584" height="327"></p>
<p>The line with</p>
<pre><code>- uses: ./
</code></pre>
<p>does the same as the one below but with no parameters given. So it defaults them.</p>
<p><strong>Still there’s the issue with undefaulted required parameters. I would expect Github to complain about this at execution time.</strong></p>
<p><a class="mention" href="/paramt">@paramt</a> I think in your case it’s quite simple to solve. You missed to hand your inputs over into your Docker Container.<br>
You’d need to adapt your <code>action.yml</code> like this (pay attention to the <code>args</code> section):</p>
<pre><code>name: 'test action'
author: Param Thakkar
description: 'test'

inputs:
  test_var:
    description: 'A test variable'
    required: true

runs:
  using: 'docker'
  image: 'Dockerfile'
  args:
    - ${{ inputs.test_var }}
</code></pre>
<p>hth<br>
Patrick</p>

**Comment by paramt** [comment, marked as answer]
<p>Yep, the problem was that the environment variable needed the ‘INPUT_’ prefix…</p>
<p>So instead of looking for the TEST_VAR environment variable, I tried INPUT_TEST_VAR and it worked.</p>
